templates/dashboard/includes/_collection_availability.html

```diff
@@ -1,9 +1,20 @@
 {% load i18n %}
 
   <div class="label label-{{ label_cls }}">
-    {% if is_published %}
+    {% if is_available %}
       {% trans "Published" context "Collection availability status" %}
-    {% else %}
-      {% trans "Not Published" context "Collection availability status" %}
-    {% endif %}
+    {% elif not collection.is_published %}
+    <div class="label">
+      {% trans "Hidden" context "Collection availability status" %}
+    </div>
+    {% elif collection.published_at %}
+  <div class="label">
+    {% blocktrans with date=collection.published_at context "Collection availability status" %}
+      Hidden (will become visible on {{ date }})
```

Comment:
The `elif` logic is wrong here. You want this code to be executed whenever `is_published is True` and `published_at is not None`.

Reply:
*partially resolved, see recent comments*